includes/functions/functions_prices.php

```diff
@@ -489,8 +493,12 @@ function zen_get_discount_calc($product_id, $attributes_id = false, $attributes_
     }
 
     $new_products_price = zen_get_products_base_price($product_id);
-    $new_special_price = zen_get_products_special_price($product_id, true);
     $new_sale_price = zen_get_products_special_price($product_id, false);
+    if ($new_sale_price !== false) {
```

Comment:
Curiously, `$new_sale_price` isn't used in this function anywhere. 
Let's revert this section (line 492) back, and then this new if/else can be removed too.